runtime/doc/lsp.txt

```diff
@@ -169,73 +170,130 @@ Requests and notifications defined by the LSP specification are referred to as
-- workspace/symbol
-- workspace/workspaceFolders
-
-                                                                 *lsp-handler*
-LSP handlers are functions that handle |lsp-response|s to requests made by Nvim
-to the server. (Notifications, as opposed to requests, are fire-and-forget:
-there is no response, so they can't be handled. |lsp-notification|)
+  handler                                    type ~
+  ------------------------------------------ -------------- ~
+  $/progress                                 notification
+  workspace/executeCommand                   response
+  window/workDoneProgress/create             request
+  window/showMessageRequest                  request
+  client/registerCapability                  request
+  client/unregisterCapability                request
+  workspace/applyEdit                        request
+  workspace/configuration                    request
+  workspace/workspaceFolders                 request
+  textDocument/publishDiagnostics            notification
+  textDocument/diagnostic                    response
+  textDocument/codeLens                      response
+  textDocument/inlayHint                     response
+  workspace/inlayHint/refresh                request
+  textDocument/references                    response
+  textDocument/documentSymbol                response
+  workspace/symbol                           response
+  textDocument/rename                        response
+  textDocument/formatting                    response
+  textDocument/rangeFormatting               response
+  textDocument/completion                    response
+  textDocument/hover                         response
+  textDocument/declaration*                  response
+  textDocument/definition                    response
+  textDocument/implementation*               response
+  textDocument/typeDefinition*               response
+  textDocument/signatureHelp                 response
+  textDocument/documentHighlight             response
+  callHierarchy/incomingCalls                response
+  callHierarchy/outgoingCalls                response
+  window/logMessage                          notification
+  window/showMessage                         notification
+  window/showDocument                        request
+  textDocument/semanticTokens/full           response
+  textDocument/semanticTokens/full/delta     response
+  workspace/semanticTokens/refresh           request
+  textDocument/codeAction
+
+Legend (see |lsp-handler|): ~
+  "response"     : LSP server sends a reponse to request made by client (Nvim)
+  "request"      : LSP server sends a request to client (Nvim)
+  "notification" : LSP server sends a notification to client (Nvim)
 
-Each response handler has this signature: >
 
-    function(err, result, ctx, config)
+                                                                 *lsp-handler*
+LSP handlers are functions executed on the Nvim's side (LSP client):
```

Comment:
I'm used to it, but it might be confusing for other users that we often exchange the role of the client and server in our docs. E.g.: In `vim.lsp.get_client_by_id` the client is the language server, not the editor.

Reply:
See also https://github.com/neovim/neovim/pull/31031#discussion_r1874415412 ; "configs" in particular overlap both client and server concepts.